featuretools/entityset/entityset.py

```diff
@@ -485,6 +485,9 @@ def entity_from_dataframe(self,
         if time_index is not None and time_index == index:
             raise ValueError("time_index and index cannot be the same value, %s" % (time_index))
 
+        if vtypes.DatetimeTimeIndex in variable_types.values() and time_index is None:
+            raise ValueError("'DatetimeTimeIndex' must be set using time_index parameter")
```

Comment:
We should include the variable name in the error message to help the user track down the source of the problem